src/main/java/com/lpnu/iot/term2/lab2/WeaponWriter.java

```diff
@@ -0,0 +1,49 @@
+package com.lpnu.iot.term2.lab2;
+
+import java.io.FileWriter;
+import java.util.ArrayList;
+import java.util.List;
+
+public class WeaponWriter {
+
+    private WeaponWriter() {
+
+    }
+
+    public static void writeToCSV(List<Weapon> weapons) {
+        var defences =  new ArrayList<DefenceWeapon>();
+        var assaults =  new ArrayList<AssaultWeapon>();
+
+        for (var weapon : weapons) {
+            try {
+                var casted =  (DefenceWeapon) weapon;
+                defences.add(casted);
+            } catch (java.lang.ClassCastException e) {
+                assaults.add(((AssaultWeapon) weapon));
+            }
+        }
+
+        try {
+            var writer = new FileWriter("weapons.csv");
+
+
+            writer.write(DefenceWeapon.getHeaders() + "\r\n");
+            for (var weapon : defences)
+                writer.write(weapon.toCSV() + "\r\n");
+
+            writer.write(AssaultWeapon.getHeaders() + "\r\n");
+            for (var weapon : assaults)
+                writer.write(weapon.toCSV() + "\r\n");
+
+            writer.close();
+        } catch (java.io.IOException e) {
+            System.out.println(
+                    "Unable to write weapons to file: "
+                            + "an error occurred.");
+            e.printStackTrace();
+        }
+
+    }
```

Comment:
Consider switching to try-with-resources here, or at least:
- move your close() operation to the `finally` block (simply because your code can throw an exception BEFORE close() operation! This is important),
- and consider adding flush() (see [this about flush](https://stackoverflow.com/questions/9858495/using-flush-before-close), we'll talk about it during our time slot).